src/appshell/view/dockwindow/dockwindow.cpp

```diff
@@ -365,6 +384,23 @@ void DockWindow::setCurrentPageUri(QString uri)
     emit currentPageUriChanged(m_currentPageUri);
 }
 
+void DockWindow::setMenues(QVariantList menues)
+{
+    MenuItemList items;
+    for (const QVariant& item: menues) {
+        items << MenuItem::fromVariantMap(item.toMap());
+    }
+
+    m_menues = items;
+
+    m_window->menuBar()->clear();
+    for (const MenuItem& menu: m_menues) {
+        m_window->menuBar()->addMenu(makeMenu(menu));
```

Comment:
I think it better to add `menuBar` property instead `menus` property. 
And add `MenuBar` component, like `DockPage`
And you can make 

Qml 
```
DockWindow {
...
     menuBar: DockMenuBar {
           items: ...

           onActionTriggered: ...
    }
}

```

And move all methods like `QMenu* DockWindow::makeMenu(const MenuItem& menuItem) const`, `QAction* DockWindow::makeAction(const MenuItem& menuItem) const` to `DockMenuBar`